crates/database/src/orders.rs

```diff
@@ -54,6 +54,14 @@ pub enum BuyTokenDestination {
     Internal,
 }
 
+/// one row in the pre_interaction table
+#[derive(Clone, Debug, Default, Eq, PartialEq, sqlx::FromRow)]
+pub struct PreInteraction {
```

Comment:
Naming nit:
```suggestion
pub struct Interaction {
```

The struct is the same for all interactions. I don't think the type name needs to indicate that its only for pre-interactions.